File: planeGame_android_java/app/src/main/java/cn/lblbc/game/sprite/Bullet.java
/**
 * 厦门大学计算机专业 | 前华为工程师
 * 专注《零基础学编程系列》  http://lblbc.cn/blog
 * 包含：Java | 安卓 | 前端 | Flutter | iOS | 小程序 | 鸿蒙
 * 公众号：蓝不蓝编程
 */
package cn.lblbc.game.sprite;

import android.graphics.Bitmap;

public class Bullet extends AutoSprite {

    public Bullet(Bitmap bitmap) {
        super(bitmap);
        setSpeed(-10);//负数表示子弹向上飞
    }
}

File: planeGame_flutter/lib/sprite/bullet.dart
/// 厦门大学计算机专业 | 前华为工程师
///  专注《零基础学编程系列》  http://lblbc.cn/blog
///  包含：Java | 安卓 | 前端 | Flutter | iOS | 小程序 | 鸿蒙
///  公众号：蓝不蓝编程
import 'package:flutter/cupertino.dart';

import 'auto_sprite.dart';

class Bullet extends AutoSprite {
  Bullet(String image) {
    this.image = image;
    this.speed = -8;
    this.size = Size(9, 21);
  }

  @override
  void update() {
    super.update();
  }

  @override
  Widget getRenderWidget() {
    return Positioned(
      top: this.y,
      left: this.x,
      child: Image.asset(
        this.image,
        width: this.size.width,
        height: this.size.height,
      ),
    );
  }
}
